### DISCUSSION TITLE
When will v3 be open?

### DISCUSSION THREAD

**Discussion by xiaolong-elf** [BlueBrain/HighFive#1019] [Questions] [answered]
When will v3 be open?

**Comment by 1uc** [comment]
We're working towards a release whenever we can steal time from other projects. Since, we'd prefer breaking changes to be rare, we have to make all changes that affect API before creating an actual release.

If you can and want to use a pre-release, the current master is the current state of `v3`, it's fully functional to the best of our knowledge, but not stable. There might be more breaking changes coming.

You can keep track of what still needs to be like so:
https://github.com/BlueBrain/HighFive/issues?q=is%3Aissue+is%3Aopen+label%3Av3

one of which is prepped (Modernize H5Easy), leaving 4 more projects to tackle.

**Reply by turbotage** [reply]
Is there any other release planned before v3? I don't know if the current master constitutes a lot of  breaking changes, but I would like to have the squeeze/reshape and read_raw functionality. Unfortunately, I have to consume this project via vcpkg. I was wondering if a 2.10 or 2.9.1 or something, was possible so that one can request a port update in vcpkg from 2.9? A release that is in the current state of master.

Great library! 😄

**Reply by 1uc** [reply]
There's nothing planned and unless we find something we must backport, we're likely not going to create another release of `v2`. The number of breaking changes is relatively low, `read_raw` being one, the CMake integration might break and removing broad-casting will likely also cause issues too. The rest is somewhat unlikely to cause problems, but they might.

I think vcpkg allows pre-releases, e.g. "cgal" seems to have `-beta1` versions. That would be something we can do quickly. Would that solve the issue?

**Reply by turbotage** [reply]
That would work! :smile:  I build manually until next release. If the next major release is very far away, a beta release would be nice. If it is like only 1-2 months, it might be unnecessary, at least for me. Maybe other people would like it :open_mouth:

**Reply by 1uc** [reply]
We released `2.10.0` which includes `read_raw`. I'll try and continue alkino's work for getting a `3.0.0-beta1` out.

**Reply by 1uc** [reply, marked as answer]
... and finally there's also a pre-release: `3.0.0-beta1`.

**Comment by xiaolong-elf** [comment]
thanks for your answer,  when I write  double** to hdf5, I found the api not support, but it's supported in v3  mentioned by The doc "migration_guide.md". Looking forward v3 release.
